starboard/android/shared/drm_system.cc

```diff
@@ -223,17 +251,49 @@ DrmSystem::~DrmSystem() {
   }
 }
 
+DrmSystem::SessionUpdateRequest::SessionUpdateRequest(
+    int ticket,
+    const char* type,
+    const void* initialization_data,
+    int initialization_data_size) {
+  JniEnvExt* env = JniEnvExt::Get();
+  j_ticket = static_cast<jint>(ticket);
+  j_init_data = env->ConvertLocalRefToGlobalRef(
+      ByteArrayFromRaw(initialization_data, initialization_data_size));
+  j_mime =
+      env->ConvertLocalRefToGlobalRef(env->NewStringStandardUTFOrAbort(type));
```

Comment:
nit: converting local to global reference is more expensive than saving |type| and |initialization_data| in the class and use local reference in Generate().

Reply:
When the request has to be deferred, we need to keep a copy of both `initialization_data` and  `type` to be used later, so in that case we can't just keep a local reference (*1). Alternatively, adding code that makes a C/C++ level copy of the array and the string for later use then would result in making two copies of the data, with the second one to convert it into a jobject for passing into the createSession java method, which is also expensive.

Keeping things as local reference to optimize performance for the case where it's not deferred results in two versions of almost similar code for doing the jni conversions and generating the request.

I choose to avoid that code duplication over optimization since the performance is extremely unlikely to be critical.

*1: https://developer.android.com/training/articles/perf-jni#local-and-global-references

Reply:
AFAICS, the conversion to a local reference would make another copy of the data during `ByteArrayFromRaw` and `NewStringStandardUTFOrAbort`, meaning the data is copied twice. That is more expensive than creation of a global reference to the data that is copied only once.